RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -899,420 +955,400 @@ class DeepTauId : public deep_tau::DeepTauBase {
-                    const auto& matchHitVar = muonMatchHitVars.at(subdet);
-                    const auto& matchLimits = muonMatchVarLimits.at(subdet);
-                    const auto& hitLimits = muonHitVarLimits.at(subdet);
-                    for(int station = MuonHitMatchV2::first_station_id; station <= MuonHitMatchV2::last_station_id; ++station) {
-                        const unsigned n_matches = hit_match.nMatches(subdet, station);
-                        const unsigned n_hits = hit_match.nHits(subdet, station);
-                        get(matchHitVar.first + station - 1) = getValueLinear(n_matches, 0, matchLimits.at(station - 1), true);
-                        get(matchHitVar.second + station - 1) = getValueLinear(n_hits, 0, hitLimits.at(station - 1), true);
-                    }
+            MuonHitMatchV2 hit_match(muons.at(index_muon));
+            static const std::map<int, std::pair<int, int>> muonMatchHitVars = {
+                { MuonSubdetId::DT, { dnn::muon_n_matches_DT_1, dnn::muon_n_hits_DT_1 } },
+                { MuonSubdetId::CSC, { dnn::muon_n_matches_CSC_1, dnn::muon_n_hits_CSC_1 } },
+                { MuonSubdetId::RPC, { dnn::muon_n_matches_RPC_1, dnn::muon_n_hits_RPC_1 } }
+            };
+
+            static const std::map<int, std::vector<float>> muonMatchVarLimits = {
+                { MuonSubdetId::DT, { 2, 2, 2, 2 } },
+                { MuonSubdetId::CSC, { 6, 2, 2, 2 } },
+                { MuonSubdetId::RPC, { 7, 6, 4, 4 } }
+            };
+
+            static const std::map<int, std::vector<float>> muonHitVarLimits = {
+                { MuonSubdetId::DT, { 12, 12, 12, 8 } },
+                { MuonSubdetId::CSC, { 24, 12, 12, 12 } },
+                { MuonSubdetId::RPC, { 4, 4, 2, 2 } }
+            };
+
+            for(int subdet : hit_match.MuonHitMatchV2::consideredSubdets()) {
+                const auto& matchHitVar = muonMatchHitVars.at(subdet);
+                const auto& matchLimits = muonMatchVarLimits.at(subdet);
+                const auto& hitLimits = muonHitVarLimits.at(subdet);
+                for(int station = MuonHitMatchV2::first_station_id; station <= MuonHitMatchV2::last_station_id; ++station) {
+                    const unsigned n_matches = hit_match.nMatches(subdet, station);
+                    const unsigned n_hits = hit_match.nHits(subdet, station);
+                    get(matchHitVar.first + station - 1) = getValueLinear(n_matches, 0, matchLimits.at(station - 1), true);
+                    get(matchHitVar.second + station - 1) = getValueLinear(n_hits, 0, hitLimits.at(station - 1), true);
                 }
             }
         }
-        checkInputs(inputs, is_inner ? "muon_inner_block" : "muon_outer_block", dnn::NumberOfInputs, grid.nCellsEta,
-            grid.nCellsPhi);
+        // checkInputs(inputs, is_inner ? "muon_inner_block" : "muon_outer_block", dnn::NumberOfInputs, grid.nCellsEta,
```

Comment:
Is there any important reason to keep this code commented out? If, so please comment on this in code otherwise remove.